integrations/cross_browsers/BrowserStack_Library.py

```diff
@@ -60,11 +61,33 @@ def get_active_session_details(self):
 
         return session_details
 
-
-    def get_session_logs(self, timeout=10):
-        "Return the session log in text format"
-        session_details = self.get_active_session_details()
-        session_log_url = session_details['logs']
-        session_log = requests.get(f'{session_log_url}',auth=self.auth,timeout=timeout).text
-
-        return session_log
+    def extract_session_id(self, session_url):
+        "Extract session id from session url"
+        import re
+        # Use regex to match the session ID, which is a 40-character hexadecimal string
+        match = re.search(r'/sessions/([a-f0-9]{40})', session_url)
+        if match:
+            return match.group(1)
+        else:
+            return None
+
+    def upload_terminal_logs(self, file_path, session_id = None, appium_test = False, timeout=30):
+        "Upload the terminal log to BrowserStack"
+        if session_id is None:
+            session_details = self.get_active_session_details()
+            session_id = session_details['hashed_id']
+        # File to be uploaded
+        files = {'file': open(file_path, 'rb')}
```

Comment:
Can you use context manager here so that it takes care of closing the file after use too?
```
with open(file_path, 'rb') as file:
    files = {'file': file}
    response = requests.post(url, auth=self.auth, files=files, timeout=timeout)
```

Reply:
Done